### PR TITLE
Make final fixes for lab 3

### PR DESCRIPTION
**Pull request by IvanPletinskiy**
* Add server-side validation for transactions
* Add account balance change on transaction
* Add server-side validation for topups
* Add balance change on topup
* Add server-side validation for withdrawals
* Add balance change on withdrawals
* Fix amount calculation on buy screen
* Fix currencies names in lists
* Add buy button for fiat account

### CODE REVIEW

**Review by Weizhau — APPROVED** [review]

File: frontend/package-lock.json [hunk 1]
```diff
@@ -1,11750 +0,0 @@
-{
-  "name": "client",
```
Comment (dawist-o) [inline frontend/package-lock.json:2]: Do not add changes from package-lock files into commits.
TODO: remove all package-lock files from remote repo

File: frontend/src/App.js [hunk 2]
```diff
@@ -11,13 +11,6 @@ function App() {
             .then((users) => setUsers(users))
     }, []);
 
-    const [accounts, setAccounts] = React.useState(null);
-    React.useEffect(() => {
-        fetch("/accounts")
-            .then((res) => res.json())
-            .then((accounts) => setAccounts(accounts))
-    }, []);
-
```
Comment (dawist-o) [inline frontend/src/App.js:20]: GJ: lets incapsulate logic into components

File: frontend/package-lock.json (hunk 1 above)
Comment (IvanPletinskiy) [reply frontend/package-lock.json:2]: Well I just have started by removing frontend package_lock

File: frontend/src/components/AccountsList.js [hunk 3]
```diff
@@ -3,71 +3,94 @@ import {getCurrencyNameById, isFiat} from "../utils/CryptoUtils";
...
+                                onClick={() => {
+                                    navigate("/topup", {
+                                        state: {
+                                            ca_id: account.ca_id,
+                                            ca_number: account.ca_number,
+                                            fk_cc_id: account.fk_cc_id
+                                        }
+                                    })
+                                }}>Пополнить</Button>
+                            <Button
```
Comment (dawist-o) [inline frontend/src/components/AccountsList.js:42]: Create component, like TopupButton(account). Just extraction and copy&paste

File: frontend/src/components/AccountsList.js [hunk 4]
```diff
@@ -3,71 +3,94 @@ import {getCurrencyNameById, isFiat} from "../utils/CryptoUtils";
...
+                                onClick={() => {
+                                    navigate("/topup", {
+                                        state: {
+                                            ca_id: account.ca_id,
+                                            ca_number: account.ca_number,
+                                            fk_cc_id: account.fk_cc_id
+                                        }
+                                    })
+                                }}>Пополнить</Button>
```
Comment (dawist-o) [inline frontend/src/components/AccountsList.js:41]: Create component, like TopupButton(account). Just extraction and copy&paste

File: frontend/src/components/AccountsList.js [hunk 5]
```diff
@@ -3,71 +3,94 @@ import {getCurrencyNameById, isFiat} from "../utils/CryptoUtils";
...
+                                        state: {
+                                            ca_id: account.ca_id,
+                                            ca_number: account.ca_number,
+                                            fk_cc_id: account.fk_cc_id,
+                                            fk_user_id: account.fk_user_id
+                                        }
+                                    });
+                                }}
+                            >Купить
+                            </Button>
```
Comment (dawist-o) [inline frontend/src/components/AccountsList.js:64]: Create component, like WithdrawalButton(account). Just extraction and copy&paste

File: frontend/src/components/AccountsList.js [hunk 6]
```diff
@@ -3,71 +3,94 @@ import {getCurrencyNameById, isFiat} from "../utils/CryptoUtils";
...
+                        </div>
```
Comment (dawist-o) [inline frontend/src/components/AccountsList.js:82]: BuyButton(account) component

File: frontend/src/routes/buy/BuyScreen.js [hunk 7]
```diff
@@ -52,7 +53,7 @@ function BuyScreen() {
             //Don't edit, we should use == and not === below
             const targetUsdValue = accounts.find((it) => it.ca_id == ca_id).usd_price
             const srcUsdValue = accounts.find((it) => it.ca_id == selectedOption.value).usd_price
-            const buyAmount = event.target.value * targetUsdValue / srcUsdValue
+            const buyAmount = event.target.value * srcUsdValue / targetUsdValue
             const targetCurrencyCode = accounts.find((it) => it.ca_id == ca_id).cc_code
             const youWillBuy = " " + buyAmount + " " + targetCurrencyCode
```
Comment (dawist-o) [inline frontend/src/routes/buy/BuyScreen.js:58]: Good place to use ur loved Utils, if it is possible and needed

File: server/src/db/repositories/AccountsRepository.js [hunk 8]
```diff
@@ -48,5 +48,18 @@ export class AccountsRepository {
         return newAccount
     }
 
+    async updateAccountBalance(accountId, newBalance) {
+        console.log("accountId:" + accountId + " newBalance: " + newBalance)
+        const results = await pool.query('UPDATE "CryptoAccount" SET ca_amount = $1 WHERE ca_id = $2', [newBalance, accountId])
 
+        console.log(await this.getAllAccounts())
```
Comment (dawist-o) [inline server/src/db/repositories/AccountsRepository.js:55]: Log function?

File: frontend/package-lock.json (hunk 1 above)
Comment (dawist-o) [reply frontend/package-lock.json:2]: @Weizhau @IvanPletinskiy  Check gitignore for those files, especially local IDE settings

**Pull request merged**